server/app/services/scheduler/filter/memory.rb

```diff
@@ -30,25 +27,57 @@ def for_service(service, instance_number, nodes)
         candidates
       end
 
+      def resolve_memory_from_stats(service, instance_number)
+        cache_key = "scheduler:mem:filter:#{service.id}-#{instance_number}"
+        memory = self.class.cache[cache_key]
+        unless memory
+          container = service.containers.to_a.find { |c| c.instance_number == instance_number }
+          if container
+            stats = container.container_stats.latest
+            if stats
+              memory = stats.memory['usage'] * 1.25
+              self.class.cache[cache_key] = memory
+            end
+          end
+        end
+        unless memory
+          memory = 256.megabytes # just a default if we cannot resolve anything
+        end
+
+        memory
+      end
+
+      # @param service [GridService]
+      # @param instance_number [Integer]
+      def fetch_service_instance(service, instance_number)
+        service.grid_service_instances.to_a.find { |i| i.instance_number == instance_number }
+      end
+
       # @param [HostNode] candidate
       # @param [Float] memory
       # @param [GridService] service
       # @param [Integer] instance_number
       def reject_candidate?(candidate, memory, service, instance_number)
-        return false if candidate.containers.service_instance(service, instance_number).first
+        return false if fetch_service_instance(service, instance_number)
```

Comment:
Wut? This check used to include the candidate host node in the condition? Now it's accepting all candidates if the service instance already exists on any node...?

Reply:
Good catch, not sure what I was thinking here.. and even worse that specs are green.